### DISCUSSION TITLE
How to create custom static virtual channel?

### DISCUSSION THREAD

**Discussion by srivatsagowda** [FreeRDP/FreeRDP#6951] [General]
Hi
I'm trying to get the data coming from a script in a server to the client side. In order to do that I need to make use of virtual channels. But I could not find the guide for implementing the virtual channels in FreeRDP.
Please help.

Thank You

**Comment by akallabeth** [comment]
1. Implement and export `VirtualChannelEntryEx` function in your library.
2. Set and initialize the functions with `pVirtualChannelInitEx`
3. handle `CHANNEL_EVENT_*` messages 

For full examples check the `channels` folder in source, each channel implementing `VirtualChannelEntryEx` can be used as a sample
